nifi-toolkit/nifi-toolkit-admin/src/main/groovy/org/apache/nifi/toolkit/admin/client/NiFiClientFactory.groovy

```diff
@@ -116,22 +121,27 @@ class NiFiClientFactory implements ClientFactory{
         return sslContext;
     }
 
-    private static class NiFiHostnameVerifier implements HostnameVerifier {
+    static class NiFiHostnameVerifier implements HostnameVerifier {
 
         @Override
         public boolean verify(final String hostname, final SSLSession ssls) {
             try {
-                for (final Certificate peerCertificate : ssls.getPeerCertificates()) {
-                    if (peerCertificate instanceof X509Certificate) {
-                        final X509Certificate x509Cert = (X509Certificate) peerCertificate;
-                        final List<String> subjectAltNames = getSubjectAlternativeNames(x509Cert);
-                        if (subjectAltNames.contains(hostname.toLowerCase())) {
-                            return true;
-                        }
+
+                final Certificate peerCertificate = ssls.getPeerCertificates()[0]
+                final X509Certificate x509Cert = CertificateUtils.convertAbstractX509Certificate(peerCertificate)
+                final String dn = x509Cert.getSubjectDN().getName().trim()
+                final String cn = dn.tokenize(",").find { cn -> cn.startsWith("CN=")}
+
+                if(StringUtils.isNoneEmpty(cn) && StringUtils.substringAfter(cn,"=").toLowerCase().equals(hostname.toLowerCase())){
```

Comment:
I think `StringUtils.isNoneEmpty(cn)` should be `StringUtils.isNotEmpty(cn)` -- probably an autocomplete issue. You can also use `equalsIgnoreCase()` rather than having to convert both strings to lowercase.